src/components/Pressable/Pressable.tsx

```diff
@@ -52,56 +54,64 @@ export default function Pressable(props: PressableProps) {
     [props.pressRetentionOffset]
   );
 
-  const pressGesture = useMemo(
-    () =>
-      Gesture.LongPress().onStart((event) => {
-        if (isPressedDown.current) {
-          props.onLongPress?.(adaptStateChangeEvent(event));
-          isPressCallbackEnabled.current = false;
-        }
-      }),
-    [props]
-  );
-
   const hoverInTimeout = useRef<number | null>(null);
   const hoverOutTimeout = useRef<number | null>(null);
 
   const hoverGesture = useMemo(
     () =>
       Gesture.Hover()
+        .manualActivation(true) // Stops Hover from blocking Native gesture activation on web
+        .cancelsTouchesInView(false)
         .onBegin((event) => {
           if (hoverOutTimeout.current) {
             clearTimeout(hoverOutTimeout.current);
           }
           if (props.delayHoverIn) {
             hoverInTimeout.current = setTimeout(
-              () => props.onHoverIn?.(adaptStateChangeEvent(event)),
+              () => props.onHoverIn?.(gestureToPressableEvent(event)),
               props.delayHoverIn
             );
             return;
           }
-          props.onHoverIn?.(adaptStateChangeEvent(event));
+          props.onHoverIn?.(gestureToPressableEvent(event));
         })
-        .onEnd((event) => {
+        .onFinalize((event) => {
           if (hoverInTimeout.current) {
             clearTimeout(hoverInTimeout.current);
           }
           if (props.delayHoverOut) {
             hoverOutTimeout.current = setTimeout(
-              () => props.onHoverOut?.(adaptStateChangeEvent(event)),
+              () => props.onHoverOut?.(gestureToPressableEvent(event)),
               props.delayHoverOut
             );
             return;
           }
-          props.onHoverOut?.(adaptStateChangeEvent(event));
+          props.onHoverOut?.(gestureToPressableEvent(event));
         }),
     [props]
   );
 
   const pressDelayTimeoutRef = useRef<number | null>(null);
+  const propagationGreenLight = useRef<boolean>(false);
+
+  // iOS only, propagationGreenLight setting occurs after other checks when in scroll-view
+  const awaitingEventPayload = useRef<PressableEvent | null>(null);
+
   const pressInHandler = useCallback(
-    (event: GestureTouchEvent) => {
-      props.onPressIn?.(adaptTouchEvent(event));
+    (event: PressableEvent) => {
+      if (handlingOnTouchesDown.current) {
+        awaitingEventPayload.current = event;
+      }
+
+      if (propagationGreenLight.current === false) {
```

Comment:
It's done for readability, I think `=== false` is much more readable, especially with refs which can be mistakenly used like this:

```js
if (propagationGreenLight)
```

In which case they are always truthy, and the linter doesn't tell you anything is wrong.

Reply:
IMHO it doesn't improve readability and I'd stick to using just `!` prefix (cc @j-piasecki)

Reply:
note - `propagationGreenLight` has been renamed to `isTouchPropagationAllowed` since

Reply:
Okay, I've  just discussed it with @j-piasecki and we agreed that we should stay consistent. In other places in this file we have for example:

```tsx
if (hoverInTimeout.current) {
  ...
}
```

so I believe we can use prefix `!` instead of comparing it to `false`, if it is just for readability sake 😅

Reply:
fixed in [64e9d50](https://github.com/software-mansion/react-native-gesture-handler/pull/2981/commits/64e9d50462b800b9787563b3ed9bceff4fa61bc8)